spec/models/manageiq/providers/embedded_terraform/automation_manager/configuration_script_source_spec.rb

```diff
@@ -0,0 +1,457 @@
+FakeTerraformRepo = Spec::Support::FakeAnsibleRepo # TODO: replace FakeAnsibleRepo with FakeTerraformRepo
+
+RSpec.describe ManageIQ::Providers::EmbeddedTerraform::AutomationManager::ConfigurationScriptSource do
+  context "with a local repo" do
+    let(:manager) do
+      FactoryBot.create(:provider_embedded_terraform, :default_organization => 1).managers.first
+    end
+
+    let(:params) do
+      {
+        :name    => "hello_world",
+        :scm_url => "file://#{local_repo}"
+      }
+    end
+
+    let(:clone_dir)          { Dir.mktmpdir }
+    let(:local_repo)         { File.join(clone_dir, "hello_world_local") }
+    let(:repo_dir)           { Pathname.new(Dir.mktmpdir) }
+    let(:repos)              { Dir.glob(File.join(repo_dir, "*")) }
+    let(:repo_dir_structure) { %w[hello_world.tf] }
+
+    before do
+      FileUtils.mkdir_p(local_repo)
+
+      repo = FakeTerraformRepo.new(local_repo, repo_dir_structure)
+      repo.generate
+      repo.git_branch_create("other_branch")
+      stub_const("GitRepository::GIT_REPO_DIRECTORY", repo_dir)
+
+      EvmSpecHelper.assign_embedded_terraform_role
+    end
+
+    # Clean up repo dir after each spec
+    after do
+      FileUtils.rm_rf(repo_dir)
+      FileUtils.rm_rf(clone_dir)
+    end
+
+    def files_in_repository(git_repo_dir)
+      repo = Rugged::Repository.new(git_repo_dir.to_s)
+      repo.ref("HEAD").target.target.tree.find_all.pluck(:name)
+    end
+
+    describe ".create_in_provider" do
+      # let(:notify_creation_args) { notification_args('creation') }
+
+      context "with valid params" do
+        it "creates a record and initializes a git repo" do
+          # expect(Notification).to receive(:create!).with(notify_creation_args)
+          # expect(Notification).to receive(:create!).with(notification_args("syncing", {}))
+
+          result = described_class.create_in_provider(manager.id, params)
+
+          expect(result).to be_an(described_class)
+          expect(result.scm_type).to eq("git")
+          expect(result.scm_branch).to eq("master")
+          expect(result.status).to eq("successful")
+          expect(result.last_updated_on).to be_an(Time)
+          expect(result.last_update_error).to be_nil
+
+          git_repo_dir = repo_dir.join(result.git_repository.id.to_s)
+          expect(files_in_repository(git_repo_dir)).to eq ["hello_world.tf"]
+        end
+
+        # NOTE:  Second `.notify` stub below prevents `.sync` from getting fired
+        it "sets the status to 'new' on create" do
+          pending # will fix later
+
+          # expect(Notification).to receive(:create!).with(notify_creation_args)
+          # expect(described_class).to receive(:notify).with(any_args).and_call_original
+          # expect(described_class).to receive(:notify).with("syncing", any_args).and_return(true)
```

Comment:
This gets the test passing:

```suggestion
          expect(described_class).to receive(:notify).with("creation", manager.id, params).and_call_original
          expect(described_class).to receive(:notify).with("syncing", manager.id, {}).and_return(true)
          result = described_class.create_in_provider(manager.id, params)
```

We should have a separate one testing the functionality of sync so it's ok to stub it here.